### PR TITLE
1378495: Do not touch OSTree Origin files.

### PR DESCRIPTION
**Pull request by cnsnyder**
This PR stops the subscription-manager ostree plugin from changing the contents of any ostree .origin files. It was historically useful but seems (based on the RFE and comments on the design[1]) to no longer be useful.

[1]: https://docs.google.com/document/d/1G1CwRlUOrfyRla0M9NdLS9c5j-8BJGCSwHnfIT0DHVw/edit

### CODE REVIEW

File: src/subscription_manager/plugin/ostree/gi_wrapper.py [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-# along with this software; if not, see
-# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
-#
-# Red Hat trademarks are not licensed under GPLv2. No permission is
-# granted to use or replicate Red Hat trademarks that are incorporated
-# in this software or its documentation.
-#
-
-"""
-Standalone script to fetch information about the current OSTree.
```
Comment (alikins) [inline src/subscription_manager/plugin/ostree/gi_wrapper.py]: :+1: Glad to see the origin updating code go away.

File: src/subscription_manager/plugin/ostree/config.py [hunk 2]
```diff
@@ -40,19 +39,19 @@
...
 except with "defaults" support removed.
 
-We has a KeyFileConfigParser, and two subclasses of it
-for RepoFileConfigParser and OriginFileConfigParser.
+We has a KeyFileConfigParser, and one subclasses of it
+for RepoFileConfigParser.
 
-There is also a OstreeConfigFile, and two subsclasses of
-it for RepoFile, and OriginFile. These add some file type
+There is also a OstreeConfigFile, and one subsclasses of
```
Comment (kahowell) [inline src/subscription_manager/plugin/ostree/config.py]: Same as above ("one subclass")

**Comment by kahowell** [comment]
Looks good to me!

**Comment by cgwalters** [comment]
Hum, is it that simple?  I think we'll have a problem unless the remote names match between what's laid down in the ISO and what's provided via subman.

ISO: `rhel-atomic-host`
subman: `rhel-atomic-host-ostree`

Did someone test this?  My guess is `atomic host upgrade => "No remote rhel-atomic-host found"`.  I suspect we'll need some logic like "only reset the origin if it matches `refspec=rhel-atomic-host:rhel-atomic-host/7/x86_64/standard`" or something like that.

One messy aspect here is that some users are going to want subman for the *rpm-md* (i.e. yum) repos, but they may want to rebase their host.  (This is mostly an internal testing scenario, not most users)

**Comment by cgwalters** [comment]
We could probably trim a lot of code by having subman talk DBus to rpm-ostree.

**Comment by cnsnyder** [comment]
Isn't that an issue with the ISOs themselves and not something that should be determined by subscription-manager? I believe we should not touch deployed origin files. Won't the deployed origin be changed by rpm-ostree/ostree when a rebase occurs? Is the issue here only that the user will not be able to run `atomic host upgrade` immediately after subscribing?

**Comment by cgwalters** [comment]
We can definitely change the ISO's remote name to match, and I *think* it would make this work.

What I'm getting at is the default primary customer path is *not* to rebase, it's just `atomic host upgrade` which consumes the existing origin.  Hence, it needs to work.

**Comment by cgwalters** [comment]
Bottom line, before this subman update lands we need to change the RHEL AH build process.  I filed a card here https://trello.com/c/7N8t3fuY/593-3-tweak-rhelah-build-process-for-new-subman-and-verify

**Comment by jlebon** [comment]
Which RHAH release will this be targeting? So that we can coordinate the build process changes.

**Comment by cgwalters** [comment]
Actually, another thing that breaks with this is the `unconfigured-state` key.  See https://github.com/ostreedev/ostree/commit/b3ad113f788eaece9a4764ecafd3308154f59e8a
for the introduction of that.

Now, we can potentially reinvent another way to do this.  I filed https://github.com/ostreedev/ostree/issues/626 for that.

I am pretty disappointed that no one tried actually testing this PR on Atomic Host before merging it, or even notified one of the AH developers about this to look at it.

Anyways, I'll keep looking at reworking things from the AH side for this.

**Comment by cnsnyder** [comment]
@cgwalters Apologies for any confusion. I had been talking to Ben Breard regarding this from the get go. It was my impression that the OStree folks were asking for this and had well reviewed the design document that was written regarding this change prior to the change.

If we'd like to take this back to the drawing board I am open to discussion. I want to get this done the right way. It is appearing ever more to me that there has been a misunderstanding here.

@jlebon This change was to be included with the next RHEL release (I'm uncertain what that means for RHAH).

**Comment by cgwalters** [comment]
I was playing with this, and in my quick testing, I'm seeing a behavior where subman isn't updating the remote config either.  I'll post an ISO early next week with reworked configs and new code so we can level set.

**Comment by cgwalters** [comment]
Nevermind, pilot error :man: :airplane: :bomb: - I just forgot to run `attach`.  Things seem to work for me!

**Pull request merged**